objects/src/notes/location.rs

```diff
@@ -1,74 +1,63 @@
 use alloc::string::ToString;
+use core::num::TryFromIntError;
 
 use super::{
-    ByteReader, ByteWriter, Deserializable, DeserializationError, Digest, NoteError, Serializable,
-    NOTE_TREE_DEPTH,
+    ByteReader, ByteWriter, Deserializable, DeserializationError, NoteError, Serializable,
 };
-use crate::crypto::merkle::{MerklePath, NodeIndex};
+use crate::crypto::merkle::MerklePath;
 
-/// Contains information about the origin of a note.
+/// Contains information about the location of a note.
 #[derive(Clone, Debug, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct NoteOrigin {
+pub struct NoteLocation {
     /// The block number the note was created in.
-    pub block_num: u32,
+    block_num: u32,
 
     /// The index of the note in the note Merkle tree of the block the note was created in.
-    pub node_index: NodeIndex, // TODO: should be a u32 because the depth is always the same
+    node_index_in_block: u32,
+}
+
+impl NoteLocation {
+    pub fn block_num(&self) -> u32 {
+        self.block_num
+    }
+
+    pub fn node_index(&self) -> u64 {
+        self.node_index_in_block as u64
+    }
 }
 
 /// Contains the data required to prove inclusion of a note in the canonical chain.
 #[derive(Clone, Debug, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 pub struct NoteInclusionProof {
-    /// Details about the note's origin.
-    origin: NoteOrigin,
-
-    /// The sub hash of the block the note was created in.
-    sub_hash: Digest,
-
-    /// The note root of the block the note was created in.
-    note_root: Digest,
+    /// Details about the note's location.
+    location: NoteLocation,
 
     /// The note's authentication Merkle path its block's the note root.
     note_path: MerklePath,
 }
 
 impl NoteInclusionProof {
     /// Returns a new [NoteInclusionProof].
-    pub fn new(
-        block_num: u32,
-        sub_hash: Digest,
-        note_root: Digest,
-        index: u64,
-        note_path: MerklePath,
-    ) -> Result<Self, NoteError> {
-        let node_index = NodeIndex::new(NOTE_TREE_DEPTH, index)
-            .map_err(|e| NoteError::invalid_origin_index(e.to_string()))?;
+    pub fn new(block_num: u32, index: u64, note_path: MerklePath) -> Result<Self, NoteError> {
         Ok(Self {
-            origin: NoteOrigin { block_num, node_index },
-            sub_hash,
-            note_root,
+            location: NoteLocation {
+                block_num,
+                node_index_in_block: index.try_into().map_err(|err: TryFromIntError| {
+                    NoteError::InvalidLocationIndex(err.to_string())
+                })?,
+            },
```

Comment:
I think we need to do more validation for `node_index_in_block` as not every `u32` value is a valid index in a depth 21 tree.